x-pack/plugins/ml/public/application/explorer/anomaly_explorer_context.tsx

```diff
@@ -70,10 +70,17 @@ export const AnomalyExplorerContextProvider: FC = ({ children }) => {
   // eslint-disable-next-line react-hooks/exhaustive-deps
   const mlResultsService = useMemo(() => mlResultsServiceProvider(mlApiServices), []);
 
-  const [anomalyExplorerContextValue, setAnomalyExplorerContextValue] =
-    useState<AnomalyExplorerContextValue>(undefined);
+  const [anomalyExplorerContextValue, setAnomalyExplorerContextValue] = useState<
+    AnomalyExplorerContextValue | undefined
+  >(undefined);
 
+  // It might look tempting to refactor this into `useMemo()` and just return
+  // `anomalyExplorerContextValue`, but these services internally might call other state
+  // updates so using `useEffect` is the right thing to do here to not get errors
+  // related to React lifecycle methods.
   useEffect(() => {
+    const explorerService = explorerServiceFactory(mlFieldFormatService);
```

Comment:
Same note here - `mlFieldFormatService` doesn't need to be pulled from context but can be instantiated in this component once (using the required services already in context) to be passed on.